### PR TITLE
Font Library: Fix focus loss when update/install button is pressed

### PR DESCRIPTION
**Pull request by t-hamano**
## What?

This PR fixes an issue where the font library loses focus when updating or installing fonts.

## Why?

Once the font is updated and installed, the button will have a disabled attribute added to it. If the button has this attribute, it is not possible to focus on the button and the button loses focus. Users who only use the keyboard may have a hard time returning focus to the Font Library.

## How?

I used `aria-disabled="true"` instead of `disabled`. This allows the button to maintain focus even if it is inactive. This is the approach also used by the Post Editors and the Site Editor.

image

Also, in the current implementation, the footer itself is not displayed when no font is selected to be installed. This PR displays the footer even when no font is selected and renders an inactive button.

### Testing Instructions for Keyboard

- Open the Font Library.
- Install or Update fonts.
- Focus will not be lost and you will be able to continue using the keyboard.

> [!NOTE]
Unlike installing fonts, updating fonts does not notify the screen reader of any information, so users cannot understand what happened. This is also an issue with trunk, so it needs to be fixed separately. Additionally, focus is also lost when the font is removed, so this also needs to be fixed separately.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/9d9602f0-5cd1-442e-aa58-30bea2bfd061

### CODE REVIEW

**Review by afercia — APPROVED** [review]
Looks good to me.

**Comment by afercia** [comment]
> the footer itself is not displayed when no font is selected to be installed. This PR displays the footer even when no font is selected and renders an inactive button.

@t-hamano nice addition, thank you. 

- updating fonts does not notify the screen reader
- focus is also lost when the font is removed

Are these two bugs already tracked in other GitHub issues?

**Comment by t-hamano** [comment]
@afercia Thanks for the review!

> - updating fonts does not notify the screen reader
> - focus is also lost when the font is removed
> 
> Are these two bugs already tracked in other GitHub issues?

As far as I know, it hasn't been reported anywhere yet. I plan to submit both problems as issues later.

**Review by afercia — CHANGES_REQUESTED** [review]
Sorry I spoke too soon.
We also need to noop the buttons when they are aria-disabled. The pattern to use is similar to the one used for many other buttons e.g. the Save button. something like 
`onClick={ (condition to disable the button) ? undefined : callback }`

File: packages/edit-site/src/components/global-styles/font-library-modal/font-collection.js [hunk 1]
```diff
@@ -299,7 +300,7 @@ function Footer( { handleInstall } ) {
 				variant="primary"
 				onClick={ handleInstall }
```
Comment (afercia) [inline packages/edit-site/src/components/global-styles/font-library-modal/font-collection.js]: The button should be noop'ed when `isDisabled || isInstalling`

File: packages/edit-site/src/components/global-styles/font-library-modal/installed-fonts.js [hunk 2]
```diff
@@ -186,7 +186,7 @@ function Footer( { shouldDisplayDeleteButton, handleUninstallClick } ) {
 				) }
 			</div>
 			<Button
-				disabled={ ! fontFamiliesHasChanges }
+				aria-disabled={ ! fontFamiliesHasChanges }
 				variant="primary"
 				onClick={ saveFontFamilies }
```
Comment (afercia) [inline packages/edit-site/src/components/global-styles/font-library-modal/installed-fonts.js]: The button should be noop'ed when `! fontFamiliesHasChanges`

**Comment by t-hamano** [comment]
> We also need to noop the buttons when they are aria-disabled. The pattern to use is similar to the one used for many other buttons e.g. the Save button. something like

Yes, I just noticed that too, so I updated it 😅

**Comment by t-hamano** [comment]
Oops, I noticed that the following issue has already been reported.

> - updating fonts does not notify the screen reader

- #54601

> - focus is also lost when the font is removed

- #58314

**Review by afercia — APPROVED** [review]
Changes look good to me 🎉 
Unrelated test is failing, will restart the job to see it everything is green,

**Comment by t-hamano** [comment]
@afercia Thanks for the review!

As seen in [this comment](https://wordpress.slack.com/archives/C02RQC4LY/p1706517911581429) on slack, it seems that the test failure also occurred on trunk. This problem is now resolved, so I merged the trunk again. All tests should now pass.

**Comment by afercia** [comment]
Thanks @t-hamano. Yes Inoticed other PRs failing today and I rebased them. I think the preferred flow in this project is to _rebase_ on top of trunk and not merge?

**Comment by t-hamano** [comment]
I don't think there is a strict rule as to whether we should use merge or rebase. It seems that different people prefer different things.

Personally, when I want to import trunk changes, I follow the following rules:

- Use "rebase" if no one has reviewed it yet
- Use "merge" if there are already comments or reviews.

If I already have a comment tree hanging in a PR, I feel like a force push will interrupt that timeline.

However, Gutenberg uses [squash merge](https://learn.microsoft.com/en-us/azure/devops/repos/git/merging-with-squash?view=azure-devops#squash-merge) by default, so no matter how many commits are made to one PR, they will be aggregated into one commit in the trunk.

So in the end, I think either one is fine 👍

**Comment by afercia** [comment]
> However, Gutenberg uses squash merge

Right, thanks.

**Comment by ciampo** [comment]
Thank you @t-hamano for working on this!

As a small follow-up, would you mind using the `Button`'s APIs instead?

Instead of using `aria-disabled`, you should be able to get the same behavior by using the `disabled` prop (like before), and adding the `__experimentalIsFocusable` prop too. Thank you 🙏

**Comment by t-hamano** [comment]
> As a small follow-up, would you mind using the `Button`'s APIs instead?

All right 👍

**Pull request merged**